tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -3,107 +3,129 @@
-static_assert(!can_construct_at<void, const volatile int&, const volatile int&>);
-
-static_assert(can_construct_at<void, X>);
-static_assert(can_construct_at<void, X, X>);
-static_assert(can_construct_at<void, X, const X&>);
-static_assert(can_construct_at<void, X, X&>);
-static_assert(can_construct_at<void, X, X&&>);
-
-static_assert(can_construct_at<void, std::string>);
-static_assert(can_construct_at<void, std::string, std::size_t, char>);
-static_assert(!can_construct_at<void, std::string, std::size_t, char, char>);
-static_assert(!can_construct_at<void, std::string, X>);
-
-static_assert(noexcept(std::construct_at(std::declval<int*>(), 42)));
-static_assert(!noexcept(std::construct_at(std::declval<std::string*>(), "hello")));
-static_assert(noexcept(std::construct_at(std::declval<const int*>(), 42)));
-static_assert(!noexcept(std::construct_at(std::declval<const std::string*>(), "hello")));
-static_assert(noexcept(std::construct_at(std::declval<volatile int*>(), 42)));
-static_assert(!noexcept(std::construct_at(std::declval<volatile std::string*>(), "hello")));
-static_assert(noexcept(std::construct_at(std::declval<const volatile int*>(), 42)));
-static_assert(!noexcept(std::construct_at(std::declval<const volatile std::string*>(), "hello")));
-
-static_assert(noexcept(std::destroy_at(std::declval<int*>())));
-static_assert(noexcept(std::destroy_at(std::declval<std::string*>())));
-static_assert(noexcept(std::destroy_at(std::declval<const int*>())));
-static_assert(noexcept(std::destroy_at(std::declval<const std::string*>())));
-static_assert(noexcept(std::destroy_at(std::declval<volatile int*>())));
-static_assert(noexcept(std::destroy_at(std::declval<volatile std::string*>())));
-static_assert(noexcept(std::destroy_at(std::declval<const volatile int*>())));
-static_assert(noexcept(std::destroy_at(std::declval<const volatile std::string*>())));
+inline constexpr bool
+    can_construct_at_impl<void_t<decltype(construct_at(declval<Ty*>(), declval<Types>()...))>, Ty, Types...> = true;
+
+template<class Ty, class... Types>
+inline constexpr bool can_construct_at = can_construct_at_impl<void, Ty, Types...>;
+
+static_assert(can_construct_at<int>);
+static_assert(can_construct_at<const int>);
+static_assert(can_construct_at<volatile int>);
+static_assert(can_construct_at<const volatile int>);
+static_assert(can_construct_at<int, int>);
+static_assert(can_construct_at<const int, int>);
+static_assert(can_construct_at<volatile int, int>);
+static_assert(can_construct_at<const volatile int, int>);
+static_assert(can_construct_at<int, int&>);
+static_assert(can_construct_at<const int, int&>);
+static_assert(can_construct_at<volatile int, int&>);
+static_assert(can_construct_at<const volatile int, int&>);
+
+struct X {};
+
+#ifndef __EDG__ // TRANSITION, VSO-1075296
+static_assert(!can_construct_at<int, X>);
+static_assert(!can_construct_at<X, int>);
+#endif // __EDG__
+
+// note that indestructible isn't constructible but is construct_at-ible:

```

Comment:
You should propose the type trait `is_construct_at_ible` for C++23.

Reply:
(╯°□°）╯︵ ┻━┻